modules/appeals_api/spec/requests/v1/notice_of_disagreements/evidence_submissions_controller_spec.rb

```diff
@@ -5,66 +5,140 @@
+          s3_object = instance_double(Aws::S3::Object)
+          allow(s3_client).to receive(:bucket).and_return(s3_bucket)
+          allow(s3_bucket).to receive(:object).and_return(s3_object)
+          allow(s3_object).to receive(:presigned_url).and_return(+'https://fake.s3.url/foo/uuid')
           post(path, params: { nod_id: notice_of_disagreement.id, headers: headers })
+
+          expect(response.status).to eq 422
+          expect(response.body).to include "'boardReviewOption' must be 'evidence_submission'"
         end
+      end
 
-        data = JSON.parse(response.body)['data']
+      context "when nod record 'auth_headers' are present" do
+        it "returns an error if request 'headers['X-VA-SSN'] and NOD record SSNs do not match" do
+          with_settings(Settings.modules_appeals_api.evidence_submissions.location,
+                        prefix: 'https://fake.s3.url/foo/',
+                        replacement: 'https://api.vets.gov/proxy/') do
+            s3_client = instance_double(Aws::S3::Resource)
+            allow(Aws::S3::Resource).to receive(:new).and_return(s3_client)
+            s3_bucket = instance_double(Aws::S3::Bucket)
+            s3_object = instance_double(Aws::S3::Object)
+            allow(s3_client).to receive(:bucket).and_return(s3_bucket)
+            allow(s3_bucket).to receive(:object).and_return(s3_object)
+            allow(s3_object).to receive(:presigned_url).and_return(+'https://fake.s3.url/foo/uuid')
+            notice_of_disagreement.update(board_review_option: 'evidence_submission')
+            headers['X-VA-SSN'] = '1111111111'
+            post(path, params: { nod_id: notice_of_disagreement.id, headers: headers })
 
-        expect(data).to have_key('id')
-        expect(data).to have_key('type')
-        expect(data['attributes']['status']).to eq('pending')
-        expect(data['attributes']['appealId']).to eq(notice_of_disagreement.id)
-        expect(data['attributes']['appealType']).to eq('NoticeOfDisagreement')
-        expect(data['attributes']['location']).to eq('http://another.fakesite.com/rewrittenpath/uuid')
+            expect(response.status).to eq 422
+            expect(response.body).to include "'X-VA-SSN' does not match"
+          end
+        end
       end
 
-      it 'raises an error if record is not found' do
-        with_s3_settings do
-          post(path, params: { nod_id: SecureRandom.uuid })
-          expect(response.status).to eq 404
-          expect(response.body).to include 'Record not found'
+      context "when nod record 'auth_headers' are not present" do
+        # if PII expunged not validating for matching SSNs
+        it 'creates the evidence submission and returns upload location' do
+          with_settings(Settings.vba_documents.location,
+                        prefix: 'https://fake.s3.url/foo/',
+                        replacement: 'https://api.vets.gov/proxy/') do
+            s3_client = instance_double(Aws::S3::Resource)
```

Comment:
This too. So much boilerplate.

Reply:
I know. We decided to take this all out from the shared piece I created in hopes of fixing the build bug